user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/GQIDMS.md

```diff
@@ -1,50 +1,77 @@
 ---
-uid: GQIDMS
+uid: GQI_GQIDMS
 ---
 
-# GQIDMS
+# GQIDMS class
 
-Available from DataMiner 10.3.4/10.4.0 onwards. <!-- RN 35701 -->
+## Definition
 
-The `GQIDMS` class contains the following methods, which can be used to request information in the form of `DMSMessage` objects:
+- Namespace: `Skyline.DataMiner.Analytics.GenericInterface`
+- Assembly: `SLAnalyticsTypes.dll`
 
-| Method | Function |
-|--------|----------|
-| `DMSMessage SendMessage(DMSMessage message)` | Sends a request that expects a single response. |
-| `DMSMessage[] SendMessages(params DMSMessage[] messages)` | Sends multiple requests at once, or sends a request that expects multiple responses. |
+Represents the DataMiner System (DMS). Can be used to request information from the system. Communication happens through DMSMessage objects.
 
-Generally, an ad hoc data source implementation will want to add a private field where it can store the `GQIDMS` object to be used later in other callbacks when columns and rows are created.
+Available from DataMiner 10.3.4/10.4.0 onwards.<!-- RN 35701 -->
+
+> [!TIP]
+> See also: [Example of retrieving data by means of DMS messages](#example-of-retrieving-data-by-means-of-dms-messages)
+
+## Methods
+
+The `GQIDMS` class contains the methods listed below. These can be used to request information in the form of `DMSMessage` objects.
 
 > [!IMPORTANT]
 > DMS messages are subject to change without notice. If you can implement an alternative using the [built-in data sources](xref:Query_data_sources), we highly recommend that you do so instead.
 
-> [!TIP]
-> See also: [Example of retrieving data by means of DMS messages](#example-of-retrieving-data-by-means-of-dms-messages)
+### DMSMessage SendMessage(DMSMessage message)
+
+Sends one request to retrieve one response from the DMS.
+
+This is a shortcut for [SendMessages](#dmsmessage-sendmessagesparams-dmsmessage-message).
+
+#### Parameters
+
+- `DMSMessage` `message`: The request message for the DMS.
+
+#### Returns
+
+A single response from the DMS or `null`. If the DMS replied with more than 1 response, the first one will be returned.
+
+### DMSMessage[] SendMessages(params DMSMessage[] message)
+
+Sends one or more messages to retrieve multiple responses from the DMS.
+
+#### Parameters
+
+- `DMSMessage[]` `messages`: The request messages for the DMS.
+
+#### Returns
+
+A `DMSMessage` array of all responses from the DMS.
 
 ## Retrieving data by means of DMS messages
```

Comment:
Right now, this procedure is a little hidden for users, as it's a practical how-to procedure tucked away in the middle of a reference section. It might be better if we moved this to a new page under "Configuring an ad hoc data source". What do you think?

Reply:
I concur that the current placement seems excessively tucked away. It's probable that users will initiate their journey with the question: 'How do I fetch data from my DMS?' Shifting it to 'Configuring an ad hoc data source' might not be ideal, as that page already serves a singular purpose in my perspective.

In my view, the 'Configuring an ad hoc data source' page stands as a fundamental starting point, a prerequisite for every tutorial. Perhaps this page could evolve into a 'Getting Started' tutorial itself. Additionally, I suggest transforming 'Retrieving data by means of DMS messages' into a comprehensive tutorial. Both this tutorial and others should be prominently featured within the 'Configuring an ad hoc data source' page/tutorial.

However, I that's probably something out of scope for this PR?